File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int max(int a,int b) {return a>b?a:b;}
    int min(int a,int b) {return a<b?a:b;}
    int maxProfit(vector<int>& prices) {
        int res=0,mp=prices[0];
        for(int p:prices){
            mp = min(mp,p);
            res = max(res,p-mp);
        }
        return res;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.rs
impl Solution {
    pub fn max(a:i32,b:i32)->i32{
        if a>b {
            a
        }else {

        b
        }
    }
    pub fn min(a:i32,b:i32)->i32{
        if a<b{
            a
        }else {
            b
        }
    }
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut res:i32 = 0;
        let mut mp:i32 = prices[0];
        for i in prices {
            mp = Solution::min(mp,i);
            res = Solution::max(res,i-mp);
        } 
        res
    }
}